packages/runtime/src/next/lib/next.ts

```diff
@@ -0,0 +1,30 @@
+// Next.js 15 compatibility layer:
+//
+// `draftMode`, `cookies` and `params` functions became async in Next.js 15
+// (https://nextjs.org/blog/next-15#async-request-apis-breaking-change),
+// but we are supporting Next.js 13, 14 and 15 in the same codebase, and are
+// compiling against Next.js 14 as a common denominator. The wrapper
+// functions below force the async use of the aforementioned primitives in
+// the rest of the codebase even though they are typed as synchronous in
+// Next.js 14. This works because awaiting a non-promise value is a no-op in
+// JavaScript.
+
+import { cookies as _cookies, draftMode as _draftMode } from 'next/headers'
+import { NextApiRequest } from 'next'
+
+export async function draftMode() {
+  return await _draftMode()
+}
+
+export async function cookies() {
+  return await _cookies()
+}
+
+export async function enableDraftMode(enable: boolean = true) {
+  const dm = await draftMode()
+  return enable ? dm.enable() : dm.disable()
+}
```

Comment:
I don't understand why we'd want this utility to exist. It seems like an unnecessary layer of abstraction

Reply:
Good question, I thought `await enableDraftMode(false)` is clearer than `;(await draftMode()).enable()`, but maybe we want to get used to the semicolon here, haha

Reply:
Hahaha, I agree that would be a lot clearer. I think in this case I'd opt for what Next.js [does in their docs](https://nextjs.org/docs/app/api-reference/functions/draft-mode#enabling-draft-mode):

```ts
const draft = await draftMode()
draft.enable()
```

This is a nit and stylistic choice, though, so don't feel like this needs to be addressed to ship this thing. I just tend to avoid abstractions that just save some LOCs or characters but I can understand their appeal in certain scenarios where my personal taste wouldn't necessarily be attracted to it.

Reply:
For me, it's more about mental overhead of parsing and understanding "usage-land" code rather than saving LOCs/ characters. "Purity" of the API is less of a concern to me personally then readability/comprehension of the code that uses it.